feat: Add revenue goals for Revenue Analytics

rafaeelaudibert opened the pull request:
Tiny new feature to make Revenue Analytics slightly cooler to use. You can set these as far back/future as you want. We then display them in the UI as goalLines in the line chart. We can do more cool stuff in the future with them, such as displaying how close you are to your goal, or maybe sending an email congratulating people if they reach it, throw some confetti if they're early in their revenue goals, etc.

<img width="1194" alt="image" src="https://github.com/user-attachments/assets/2780e2ed-3ae7-46af-8a92-d893a55ceecc" />

<img width="942" alt="image" src="https://github.com/user-attachments/assets/81dea653-ed95-4ef1-85f9-3567b2415786" />

    
 No fun stuff for now, just useful :)

rafaeelaudibert committed:
feat: Add revenue goals for Revenue Analytics

Tiny new feature to make Revenue Analytics slightly cooler to use. You can set these as far back/future as you want. We then display them in the UI as goalLines in the line chart. We can do more cool stuff in the future with them, such as displaying how close you are to your goal, or maybe sending an email congratulating people if they reach it, throw some confetti if they're early in their revenue goals, etc.

No fun stuff for now, just useful :)

rafaeelaudibert committed:
feat: Display form in edit mode for snapshots

rafaeelaudibert committed:
feat: Guarantee snapshot is properly rendered

lricoy reviewed (approved):
Hey, this is a fantastic feature, congrats!

No notes really

rafaeelaudibert committed:
refactor: Add nullable column

CI wont let us add an actual column even though this is fine because that table can be easily locked. Our CI isnt as permissive as it should be,

rafaeelaudibert committed:
fix: Reintroduce `save` function

rafaeelaudibert committed:
fix: Add missing snapshot

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

jabahamondes reviewed (approved):
LGTM 🚀  just bringing up that I saw some snapshots losing the empty state "Create your first revenue source", not sure if it's intended

rafaeelaudibert commented on frontend/src/lib/api.mock.ts:
```diff
@@ -144,6 +144,23 @@ export const MOCK_DEFAULT_TEAM: TeamType = {
+                goal: 1_000_000,
+            },
+            {
+                due_date: '2035-12-31', // Very in the future to avoid flappy snapshots until 2035, assuming I'll be a multimillionaire by then and wont have to handle this
```
It will not fail, but the snapshot will update :)

rafaeelaudibert committed:
Update products/revenue_analytics/frontend/revenueAnalyticsLogic.ts

Co-authored-by: graphite-app[bot] <96075541+graphite-app[bot]@users.noreply.github.com>

rafaeelaudibert committed:
fix: Fix error message

Per graphite comment

rafaeelaudibert committed:
fix: Add missing comma

for fuck's sake

rafaeelaudibert committed:
ci: Add missing mocks

rafaeelaudibert committed:
meh: Remove broken test

Cant figure out how to solve it, remove it for now

rafaeelaudibert committed:
ci: Guarantee we always run Storybook

This is a required step, but Github is dumb and because it's required it NEEDS to run. It can't be entirely skipped. So let's run the first step of each job and then skip the rest.
